File: src/main/java/io/github/seggan/jyxal/compiler/wrappers/Function.java
package io.github.seggan.jyxal.compiler.wrappers;

import org.objectweb.asm.ClassWriter;

public class Function extends JyxalMethod {

    Function(ClassWriter cw, int access, String name, String desc) {
        super(cw, access, name, desc);

        stackVar = 0;
        ctxVar = 1;

        loadStack();
        visitMethodInsn(
                INVOKESTATIC,
                "runtime/list/JyxalList",
                "create",
                "(Ljava/util/Collection;)Lruntime/list/JyxalList;",
                false
        );
        visitVarInsn(ASTORE, ctxVar);
    }
}

File: src/main/kotlin/io/github/seggan/jyxal/compiler/wrappers/Function.kt
package io.github.seggan.jyxal.compiler.wrappers

import org.objectweb.asm.ClassWriter
import org.objectweb.asm.Opcodes.ASTORE
import org.objectweb.asm.Opcodes.INVOKESTATIC

class Function internal constructor(cw: ClassWriter, access: Int, name: String, desc: String) : JyxalMethod(cw, access, name, desc) {
    init {
        stackVar = 0
        ctxVar = 1
        loadStack()
        visitMethodInsn(
            INVOKESTATIC,
            "runtime/list/JyxalList",
            "create",
            "(Ljava/util/Collection;)Lruntime/list/JyxalList;",
            false
        )
        visitVarInsn(ASTORE, ctxVar)
    }
}